### ISSUE TITLE
DOUBTS - Compatibility List and About Other Games

### ISSUE THREAD

**Issue by SilentMRG** [hrydgard/ppsspp#14349]
Hello everybody! \o/

I don't know if this is the right place to post my doubts... If not, I'm sorry! Is that I tried to open a topic in the forum and I couldn't. =( I also faced problems creating the registration. :(

I am looking for a list of compatibility of the games, obviously, that work or not in PPSSPP. I saw a list on the forum, but it seems to be out of date. Is there a current list?

Another question I have is about the Driver 76 game, is it running 100% on DirectX 9 in version 1.11.3? I use the emulator on Windows 7 Ultimate 64 Bit. I know that Resistance - Retribution, as well as Syphon Filter - Dark Mirror and Logan's Shadow, are still problematic, even after countless emulator updates. I also saw on the forum that Crazy Taxi - Fare Wars is 100% playable, is that true? I remember that in 2016 this game only featured the main menu, everything else was broken. Still in 2016, Driver 76 had numerous bugs... And then in 2018 it had crashes and graphical flaws, in addition to very low fps. After that, I left the emulator aside... And I went to live... XD... These days I went back to testing the emulator and some games... The compatibility in the games I tested is incredible! And I want to take this opportunity to congratulate everyone involved in the project! You are legends! = D

Again, I'm sorry if this is not the right place! It's just that I'm new here.

Thank you very much for all your attention! And thanks for reading. =)

**Comment by unknownbrackets** [comment]
> I don't know if this is the right place to post my doubts... If not, I'm sorry! Is that I tried to open a topic in the forum and I couldn't. =( I also faced problems creating the registration. :(

It has some annoyingly strict spam restrictions because of the constant attacks from spammers.  What error are you getting?  This type of question is normally more appropriate there.

> I am looking for a list of compatibility of the games, obviously, that work or not in PPSSPP. I saw a list on the forum, but it seems to be out of date. Is there a current list?

There are well over 1000 games (even accounting for region duplicates), so it's hard for anyone to maintain a perfect list.  You should also look here, which has a list of the most recent reported compatibility from within PPSSPP itself:
https://report.ppsspp.org/games

That said, if you're looking for a perfect and up to date list of every PSP game released and how well it runs on the latest stable version (not to mention the latest git build)... well, let me know if you find it, I'd like to see it too.

> Another question I have is about the Driver 76 game, is it running 100% on DirectX 9 in version 1.11.3? I use the emulator on Windows 7 Ultimate 64 Bit.

I'm not sure what hardware you're using, but if you can avoid Direct3D 9, I recommend another backend.  Direct3D 9 has API limitations that make emulating certain game features necessarily hard or slow (compared to OpenGL / Direct3D 11 / Vulkan), so it's the worst graphics backend right now and least likely to see improvements.

Anyway, for this game there's a workaround - #12054.  It's not fixed yet.

> I know that Resistance - Retribution, as well as Syphon Filter - Dark Mirror and Logan's Shadow, are still problematic, even after countless emulator updates.

IIRC, the latter two are depth related.  You may never see a fix on Direct3D 9 for those issues.

> I also saw on the forum that Crazy Taxi - Fare Wars is 100% playable, is that true? I remember that in 2016 this game only featured the main menu, everything else was broken.

It ended up being a thread scheduling issue, fixed in #13368.  One line of code to fix, but it sure took a lot of time to figure out that was the problem.

> The compatibility in the games I tested is incredible! And I want to take this opportunity to congratulate everyone involved in the project! You are legends! = D

Thanks.  This is in no small part because of contribution from well over 100 people, especially testing, debugging, and isolating issues.  But there are only more issues in the GitHub Issues tracker today than in 2018.  All we can do is keep fixing 'em and making other improvements.

-[Unknown]

**Comment by MojoJojoDojo** [comment]
> I am looking for a list of compatibility of the games, obviously, that work or not in PPSSPP. I saw a list on the forum, but it seems to be out of date. Is there a current list?

I and some others also helped create a compat list for multiplayer games on PPSSPP :
https://docs.google.com/spreadsheets/d/1FTPprUvf551f0aoSpQcmb_qfPP6Ef09ePraXsLbIIx4/edit#gid=0
In a roundabout way they may also may tell if the games work as intended as well.

**Comment by SilentMRG** [comment]
> It has some annoyingly strict spam restrictions due to constant spammer attacks. what error are you having Usually, this type of question is more appropriate there.

I understood. The error is 504, but sometimes 502 appears. I believe it is my internet. I do not know for sure.

> There are well over 1000 games (even counting region duplicates), so it is difficult for anyone to maintain a perfect list. You should also look here, which contains a list of the most recently reported compatibility in the PPSSPP itself:
> https://report.ppsspp.org/games

I understood. I'm not looking for a perfect list, that would be impossible! hehe However, the list you went through seems to be more up to date than the one on the forum. Thank you!

> That said, if you're looking for a perfect, up-to-date list of all the PSP games released and how it works in the latest stable version (not to mention the latest git build) ... well, tell me if you find it, I'd also like to to see.

In the case of an emulator in constant development... It will be impossible for me to find a perfect list. I understand your point of view.

> I'm not sure what hardware you're using, but if you can avoid Direct3D 9, I recommend another backend. Direct3D 9 has API limitations that make emulation of certain game features necessarily difficult or slow (compared to OpenGL / Direct3D 11 / Vulkan), so it is the worst graphics backend at the moment and least likely to see improvements.

I'm using:

I5 2.67 GHz
4 GB DDR3
Intel HD Graphics 1000

Unfortunately, I'm stuck on D3D 9, because OpenGL is slow in all games and D3D 11 has graphic errors as well as slowness. It is a pity that the D3D 9 is not supported, because for low performance computers it is the best option! In GOWs I get 30 to 60 fps, and many other games run at constant 60 fps, for example: Hellboy. By the way, what version of OpenGL does the emulator use? My onboard card supports version 3.0, and it runs smoothly on many PC games that require that version.

> Anyway, for this game there is an alternative solution - [# 12054] (https://github.com/hrydgard/ppsspp/issues/12054). It hasn't been fixed yet.

It is a pity that it has not been fixed... This is one of the best games on the PSP. I thank you for the workaround.

> IIRC, the last two are related in depth. You may never see a fix in Direct3D 9 for these problems.

What a pity! =(
Without asking too much, please look at people who use cave technology, like me and many others. It might be a good idea to add the DirectX 10.1 or DirectX 9c or 9.1 API, anyway, I don't know.

> It ended up being a thread scheduling issue, fixed in [# 13368] (https://github.com/hrydgard/ppsspp/pull/13368). One line of code to fix, but it sure took a long time to figure out what the problem was.

I understood. Glad it was corrected! I'm glad that this great title is working smoothly. =D

> Thanks. This is largely due to the contribution of well over 100 people, mainly in testing, debugging and problem isolation. But there are only more problems with the GitHub problem tracker today than in 2018. All we can do is continue to fix them and make other improvements.

You do not have to thank me. I thank you for giving us the opportunity to play our PSP games on a top quality emulator in every way! My PSP broke and it was PPSSPP that saved my games! You saved my fun! My days and those of many are happier! Thank you very much... I warmly thank each of you! May God bless each one of you!

**Comment by SilentMRG** [comment]
> I and some others also helped create a compat list for multiplayer games on PPSSPP :
> https://docs.google.com/spreadsheets/d/1FTPprUvf551f0aoSpQcmb_qfPP6Ef09ePraXsLbIIx4/edit#gid=0
> In a roundabout way they may also may tell if the games work as intended as well.
> The list also tracks git versions sometimes like Unknown wanted lol.

Thank you for the tip! Great job. :)

**Comment by SilentMRG** [comment]
That's it for today! My "problem" has been solved.

**Comment by unknownbrackets** [comment]
> I understood. The error is 504, but sometimes 502 appears. I believe it is my internet. I do not know for sure.

Hm, those sound like server errors and not internet issues.  Is the username you tried to register with the same (SilentMRG)?

> > I'm not sure what hardware you're using, but if you can avoid Direct3D 9, I recommend another backend. Direct3D 9 has API limitations that make emulation of certain game features necessarily difficult or slow (compared to OpenGL / Direct3D 11 / Vulkan), so it is the worst graphics backend at the moment and least likely to see improvements.
> 
> I'm using:
> 
> I5 2.67 GHz
> 4 GB DDR3
> Intel HD Graphics 1000
> 
> Unfortunately, I'm stuck on D3D 9, because OpenGL is slow in all games and D3D 11 has graphic errors as well as slowness. It is a pity that the D3D 9 is not supported, because for low performance computers it is the best option! In GOWs I get 30 to 60 fps, and many other games run at constant 60 fps, for example: Hellboy. By the way, what version of OpenGL does the emulator use? My onboard card supports version 3.0, and it runs smoothly on many PC games that require that version.

Ah, I see.  The emulator can use any version of OpenGL between 2.0 and 4.6, but 3.3+ is best.  Maybe there's something we could optimize there.

By the way, have you seen this?  It might improve your FPS.  https://forums.ppsspp.org/showthread.php?tid=22159

> > IIRC, the last two are related in depth. You may never see a fix in Direct3D 9 for these problems.
> 
> What a pity! =(
> Without asking too much, please look at people who use cave technology, like me and many others. It might be a good idea to add the DirectX 10.1 or DirectX 9c or 9.1 API, anyway, I don't know.

It's a lot of work to support another API, but we're already enabling various extensions for 9 (i.e. 9.1, etc.)  The trouble is that developers did things on the PSP that they just wouldn't do on a PC, to squeeze all the performance out of it.  Those things aren't directly possible on PCs - so we have to use tricks to simulate.  They can be slow, especially when the API doesn't have certain features.

-[Unknown]

**Comment by SilentMRG** [comment]
> Hm, those sound like server errors and not internet issues. Is the username you tried to register with the same (SilentMRG)?

Got it... I'm bad at these things.
Worse than I even remember which name I used! hehehe... I will try to create another profile, but, it will not be today. Anyway, thanks again for the support!

> Ah, I see. The emulator can use any version of OpenGL between 2.0 and 4.6, but 3.3+ is best. Maybe there's something we could optimize there.

> By the way, have you seen this? It might improve your FPS. https://forums.ppsspp.org/showthread.php?tid=22159

I understood. Strange that it is slow in version 3.0, because this version is the last and native of the integrated card I have. If you can optimize... It will be greatly appreciated!

Interesting this solution. I'll try... Thanks!

> It's a lot of work to support another API, but we're already enabling various extensions for 9 (i.e. 9.1, etc.) The trouble is that developers did things on the PSP that they just wouldn't do on a PC, to squeeze all the performance out of it. Those things aren't directly possible on PCs - so we have to use tricks to simulate. They can be slow, especially when the API doesn't have certain features.

Hmm... I didn't know about those limitations. Apparently, the developers took milk from stone, that's what I understood.
But if it is possible to guarantee support for DirectX 9.1 or any other version of it, it will also be greatly appreciated! Potato notebook owners like mine will be happy! XD D3D 9 is our only option. Unfortunately...

**Comment by SilentMRG** [comment]
Hello again!

I am thinking of playing these games:

The Lord of the Rings - Aragorn's Quest (USA);

Untold Legends - Brotherhood of the Blade (USA);

Untold Legends - The Warrior's Code (USA);

However, I don't want to run into games that don't work on the emulator. So, I took a look at the compatibility list, and there I could see that the emulation of the three games is marked as "Perfect", is that true? Are they working 100%?

I saw somewhere that the game Aragorn's Quest has an error when saving, and that the song does not play. I also saw, I think here and on the forum, that the other two games have errors with letters that do not appear and other graphic errors of characters that disappear, in addition to low FPS.

Thank you for your help!
